src/TextInput.jsx

```diff
@@ -16,18 +16,20 @@ class TextInput extends React.Component {
 
 	onChange(e) {
 		this.setState({ value: e.target.value });
+		this.props.onChange(e);
```

Comment:
if it's not required, `this.props.onChange` might be `undefined` - make sure you have a unit test with and without an `onChange` prop